JPetMC.cpp

```diff
@@ -21,20 +21,30 @@
 #include <G4UIExecutive.hh>
 #include <G4INCLRandom.hh>
 #include <G4UImanager.hh>
-#include <Randomize.hh>
-#include <time.h>
+#include <chrono>
+#include <thread>
+#include "Info/EventMessenger.h"
+#include <fstream> 
 
-int main (int argc, char** argv)
+void setRandomSeed()
 {
-  CLHEP::HepRandom::setTheEngine(new CLHEP::RanecuEngine());
-  G4long seed = time(NULL);
-  CLHEP::HepRandom::setTheSeed(seed);
   long seeds[2];
-  time_t systime = time(NULL);
-  seeds[0] = (long) systime;
-  seeds[1] = (long) (systime * G4UniformRand());
+  seeds[0] =  std::hash<std::thread::id>()(std::this_thread::get_id());
```

Comment:
I don't know if this is enough. This function returns thread ID and it's unique for multiple threads within a program - but I don't know if it's different between multiple executions of the same binary file. 

Better solution would be to use function getpid() to have unique number for every execution (ar PID and time in case process can get the same PID as process that was finished before). 
Even better would be to calculate srand(getpid()).

Or, since we are using ROOT we can maybe copy unique seed generation via TUUID exactly like it's done in TRandom?

Reply:
Yes, it is. Get getpid returns the caller's thread ID. 
In updated version there is a mixture of time and id